BoostTestAdapter/Settings/BoostTestInternalSettings.cs

```diff
@@ -0,0 +1,91 @@
+using System.Xml.Serialization;
+using Microsoft.VisualStudio.TestWindow.Extensibility;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+using System.Xml.XPath;
+
+namespace BoostTestAdapter.Settings
+{
+    public static class BoostTestSettingsConstants
+    {
+        public const string InternalSettingsName = "BoostTestInternalSettings";
+    }
+
+    [Export(typeof(IRunSettingsService))]
+    [SettingsName(BoostTestSettingsConstants.InternalSettingsName)]
+    public class RunSettingsService : IRunSettingsService
+    {
+        public string Name => BoostTestSettingsConstants.InternalSettingsName;
+
+        public RunSettingsService()
+        {
+        }
+
+        public IXPathNavigable AddRunSettings(IXPathNavigable runSettingDocument,
+            IRunSettingsConfigurationInfo configurationInfo, ILogger logger)
+        {
+            XPathNavigator runSettingsNavigator = runSettingDocument.CreateNavigator();
+            Debug.Assert(runSettingsNavigator != null, "userRunSettingsNavigator == null!");
+            if (!runSettingsNavigator.MoveToChild(Constants.RunSettingsName, ""))
+            {
+                return runSettingsNavigator;
+            }
+
+            var settingsContainer = new RunSettingsContainer();
+
+            runSettingsNavigator.MoveToChild(Constants.RunSettingsName, "");
+            runSettingsNavigator.AppendChild(settingsContainer.ToXml().CreateNavigator());
+
+            runSettingsNavigator.MoveToRoot();
+            return runSettingsNavigator;
+        }
+    }
+
+    [XmlRoot(BoostTestSettingsConstants.InternalSettingsName)]
+    public class RunSettingsContainer : TestRunSettings
+    {
+        public RunSettingsContainer()
+            : base(BoostTestSettingsConstants.InternalSettingsName)
+        {
+            VSProcessId = Process.GetCurrentProcess().Id;
+        }
+
+        public int VSProcessId { get; set; }
+
+        public override XmlElement ToXml()
+        {
+            var document = new XmlDocument();
+            using (XmlWriter writer = document.CreateNavigator().AppendChild())
+            {
+                new XmlSerializer(typeof(RunSettingsContainer))
+                    .Serialize(writer, this);
+            }
+            return document.DocumentElement;
+        }
+    }
+
+    [Export(typeof(ISettingsProvider))]
+    [SettingsName(BoostTestSettingsConstants.InternalSettingsName)]
+    public class RunSettingsProvider : ISettingsProvider
+    {
+        public string Name => BoostTestSettingsConstants.InternalSettingsName;
+
+        public int VSProcessId { get; set; }
+
+        public void Load(XmlReader reader)
+        {
+            if (reader.Read() && reader.Name.Equals(this.Name))
+            {
+                var serializer = new XmlSerializer(typeof(RunSettingsContainer));
+                RunSettingsContainer settings = serializer.Deserialize(reader) as RunSettingsContainer;

```

Comment:
Can this coming from a .runsettings file? Would it not be a problem that one can inject a bogus PID through that? Also the policy is to do XML validation through schema for anything coming in from outside.

Reply:
I suppose someone could specify their own pid and create their own pipe, but I can't see them doing anything bad after that. All we do is read the debugging properties from it, and set the WorkingDirectory and Environment from it. I suppose it's the same as reading any data in through the xml. Also, I'll validate through the schema. Thanks!